### PR TITLE
Fetch name from Dataservlet.java

### PR DESCRIPTION
**Pull request by RolandNaijuka**
Change the script.js and index.html and fetch username from the server (dataservlet.java)

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -12,6 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+function getUserName(){
+    const userName =  fetch("/data").then(response => response.text()).then(name);
```
Comment (cwlarson-google) [inline portfolio/src/main/webapp/script.js:16]: This doesn't look like correct syntax to me - the final `then()` either returns nothing, or exactly what it receives as an input

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -24,22 +28,12 @@ function generateRandomGreeting(){
...
-  greetingContainer.innerText = generateRandomGreeting();
-}
-
 /**
  * Change the innerHTML to a greeting
  * every time use loads or refreshes
  */
 function onLoadWindow(){
-    document.getElementById("welcome-note").innerHTML = generateRandomGreeting().concat(" My name is Roland");
+    document.getElementById("welcome-note").innerHTML = generateRandomGreeting().concat(" My name is ", getUserName());
```
Comment (cwlarson-google) [inline portfolio/src/main/webapp/script.js:36]: A way to make this a little more readable:

```
document.getElementById("welcome-note").innerHTML = `${generateRandomGreeting()} My name is ${getUserName()}`;
```

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (RolandNaijuka) [reply portfolio/src/main/webapp/script.js:16]: > This doesn't look like correct syntax to me - the final `then()` either returns nothing, or exactly what it receives as an input

Thank you for catching this error. I will correct it ASAP

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (RolandNaijuka) [reply portfolio/src/main/webapp/script.js:36]: > A way to make this a little more readable:
> 
> ```
> document.getElementById("welcome-note").innerHTML = `${generateRandomGreeting()} My name is ${getUserName()}`;
> ```

This is a helpful shortcut on how to concatenate. I will start applying it ASAP

**Pull request merged**